ref_builder/cli/utils.py

```diff
@@ -0,0 +1,16 @@
+from uuid import UUID
+
+from ref_builder.resources import RepoOTU
+from ref_builder.utils import IsolateName, IsolateNameType
+
+
+def get_isolate_id_from_key(otu: RepoOTU, isolate_key: str):
+    try:
+        return UUID(isolate_key)
+    except ValueError:
+        parts = isolate_key.split(" ")
```

Comment:
Some isolate names are like "Isolate Japan 10-3", which means you'd have a three item list from `split`.

If this becomes necessary, I think it might be better to use the capabilities of SQLite to search isolates by name. Either way, the current implementation relies on a user-supplied name matching the name in the repo exactly.

IMO, users should not be allowed to mutate the repo based on anything but the UUIDs or shortened UUID.